app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -222,3 +235,13 @@ export const updateListenSignalData = (payload, id) => {
     ],
   });
 };
+
+export const deleteListenSignalData = (id) => {
```

Comment:
I'd recommend passing the entire signal object here. Reason being is that you're going to need to delete all associated responses and promotional tweets. If you pass the listen signal object as a meta prop, you can easily removes them in their respective reducer functions.